pkg/kvstore/etcd.go

```diff
@@ -1002,8 +1037,13 @@ reList:
 		default:
 		}
 
-		e.limiter.Wait(ctx)
+		lr, err := e.limiter.Wait(ctx)
+		if err != nil {
+			continue
+		}
 		kvs, revision, err := e.paginatedList(ctx, scopedLog, w.Prefix)
+		// Using lr.Error for convenience, as it matches lr.Done() when err is nil
```

Comment:
err is alway nil here though, right?

Reply:
Not necessarily, `e.paginatedList` might return an error if one of the Get operations failed.